tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/FirebaseSubscriptionGetMethod.scala

```diff
@@ -45,13 +49,21 @@ class FirebaseSubscriptionGetMethod @Inject()()(val serializer: FirebaseSubscrip
           .map(invocationResult => InvocationWithContext(invocationResult, invocation.processingContext)))
 
   private def retrieveFirebaseSubscription(username: Username, ids: Option[FirebaseSubscriptionIds]): SFlux[FirebaseSubscription] =
-    ids match {
+    (ids match {
       case None => SFlux(repository.list(username))
       case Some(value) =>
         SFlux(repository.get(username,
           value.list.map(id => FirebaseSubscriptionId.liftOrThrow(id.id.value).toOption)
             .filter(_.isDefined)
             .map(_.get)
             .toSet.asJava))
-    }
+    })
+      .flatMap(subscription => {
+        if (isInThePast(subscription.expires, clock)) {
+          SMono(repository.revoke(username, subscription.id))
+            .`then`(SMono.empty)
+        } else {
+          SMono.fromCallable(() => subscription)
```

Comment:
```suggestion
          SMono.just(subscription)
```

Reply:
`SMono.just` is an eager evaluation. I want a lazy evaluation here.

Reply:
Isnt subscription already evaluated?

Reply:
I mean the `SMono` variable would be eagerly created even if the flow does not fall into the else block, which is useless, no?

Reply:
I think there is not difference for serving a pojo between the two.